### PR TITLE
[improvements] Fix code quality issues

### PR DESCRIPTION
**Pull request by withshubh**
### Description
Hi :wave: I ran the [DeepSource](https://deepsource.io) static analyzer on the forked copy of this repo and found some [interesting code quality issues](https://deepsource.io/gh/withshubh/mastodon/issues/). This PR fixes a few of them.

### Summary of fixes
- Removed bad use of `alias`
- Fixed operand order in the binary expression
- Prefixed unused method arguments with an underscore
- Replaced the old OpenSSL algorithmic constants with the newer strings initializers.
- Removed unnecessary UTF-8 encoding comment
- Added .deepsource.toml

### CODE REVIEW

**Review by justtryintolive — APPROVED** [review]

File: config/initializers/fast_blank.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 if String.method_defined?(:blank_as?)
   class String
-    alias_method :blank?, :blank_as?
+    alias blank? blank_as?
```
Comment (ThibG) [inline config/initializers/fast_blank.rb:3]: Any specific reason to prefer the latter?

File: spec/lib/activitypub/linked_data_signature_spec.rb [hunk 2]
```diff
@@ -81,6 +81,6 @@ def sign(from_account, options, document)
     options_hash   = Digest::SHA256.hexdigest(canonicalize(options.merge('@context' => ActivityPub::LinkedDataSignature::CONTEXT)))
     document_hash  = Digest::SHA256.hexdigest(canonicalize(document))
     to_be_verified = options_hash + document_hash
-    Base64.strict_encode64(from_account.keypair.sign(OpenSSL::Digest::SHA256.new, to_be_verified))
+    Base64.strict_encode64(from_account.keypair.sign(OpenSSL::Digest.new('SHA256'), to_be_verified))
```
Comment (ThibG) [inline spec/lib/activitypub/linked_data_signature_spec.rb:84]: Any specific reason for that change?

**Review by ClearlyClaire — COMMENTED** [review]
Hi! Thank you for your PR. Interesting tool, I will have a more detailed look at its findings later!

I have added a few questions as inline comments.

File: config/initializers/fast_blank.rb (hunk 1 above)
Comment (withshubh) [reply config/initializers/fast_blank.rb:3]: Yes! [here's](https://deepsource.io/gh/withshubh/mastodon/issue/RB-ST1002/description) the detailed description that why we should use the latter

File: spec/lib/activitypub/linked_data_signature_spec.rb (hunk 2 above)
Comment (withshubh) [reply spec/lib/activitypub/linked_data_signature_spec.rb:84]: [Here's](https://deepsource.io/gh/withshubh/mastodon/issue/RB-LI1083/description) the reason why this is an anti-pattern.

Comment (ThibG) [reply spec/lib/activitypub/linked_data_signature_spec.rb:84]: weirdly enough, I can't find anything about this supposed deprecation, the documentation still suggests using those classes, etc.
however, the change shouldn't hurt either, behind the hoods those classes just call `OpenSSL::Digest.new(name)` anyway

**Review by ClearlyClaire — APPROVED** [review]

File: spec/lib/activitypub/linked_data_signature_spec.rb (hunk 2 above)
Comment (withshubh) [reply spec/lib/activitypub/linked_data_signature_spec.rb:84]: Well, Yes. The latter is the preferred way of writing as it improves the readability.

**Pull request merged**